### PR TITLE
fix #11 and #12

### LINKED ISSUE #11: byElement seems to be missing some slicing primitives
e.g.

``` D
    int[] a = [1,2,3,4];
    auto s = a.sliced(2,2);
    s.byElement[0 .. 3] = 3; // Error
    s.byElement[] = [3,3,3,3]; // Error
```

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

**Comment by 9il** [comment]
ping @John-Colvin

File: source/mir/ndslice/selection.d [hunk 1]
```diff
@@ -1187,11 +1226,14 @@ pure nothrow unittest
...
-@safe @nogc pure nothrow unittest
+@nogc nothrow unittest
 {
     import mir.ndslice.slice;
-    import std.range: iota;
-    auto elems = 20.iota.sliced(4, 5).byElement;
+    import std.array: array;
+    import std.algorithm.comparison: equal;
+    import std.range: iota, repeat;
+    static data = 20.iota.array;
```
Comment (John-Colvin) [inline source/mir/ndslice/selection.d:1235]: why `static`?

File: source/mir/ndslice/selection.d [hunk 2]
```diff
@@ -947,8 +947,8 @@ auto byElement(size_t N, Range)(auto ref Slice!(N, Range) slice)
                 }
             }
 
-            static if (PureN == 1 && rangeHasMutableElements && !hasAccessByRef)
-            auto front(DeepElemType elem) @property
+            static if (PureN == 1 && isMutable!DeepElemType && !hasAccessByRef)
+            auto front()(E elem) @property
```
Comment (John-Colvin) [inline source/mir/ndslice/selection.d]: Why does this have to be a template? (I vaguely remember having to do this myself, but I can't remember why...)

File: source/mir/ndslice/selection.d (hunk 1 above)
Comment (9il) [inline source/mir/ndslice/selection.d:1235]: `@nogc` will work only with static

Comment (John-Colvin) [inline source/mir/ndslice/selection.d:1235]: ok, I understand.

File: source/mir/ndslice/selection.d (hunk 2 above)
Comment (9il) [inline source/mir/ndslice/selection.d]: Thanks, fixed

**Comment by 9il** [comment]
implemented in `v0.10.2`

**Pull request merged**